### PR TITLE
Initial structure

### LINKED ISSUE #1: Initial Django App setup
Let's use [MyST/Sphinx](https://myst-parser.readthedocs.io/en/latest/index.html) for documentation.

### PR DESCRIPTION
**Pull request by xmnlab**
This PR adds initial structure for the django-scrud application:

- resource and resource type models and rest api
- tests
- documentation using myst and sphinx
- CI for documentation publishing and app tests

current rest api and tests is very simple, it is just for having an initial structure. I will work in a follow up PR in the real implementation.

### CODE REVIEW

**Comment by xmnlab** [comment]
@dcharbon this PR is ready for review.

feel free to "squash and merge" when you agree with this changes (I don't need the commits history).

the CI is not working yet because it needs to be merged first

the schema and json-ld implementation will be addressed in a follow up PR.

thanks!

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: "3.8"
+services:
+    scrud_db:
+      image: "mdillon/postgis:10"
```
Comment (dcharbon) [inline docker/docker-compose.yml]: Do we need a GIS capable postgres? Would a vanilla postgres image suffice?

File: scrud_django/tests/test_rest.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import json
+import os
+from unittest import skip
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
```
Comment (dcharbon) [inline scrud_django/tests/test_rest.py]: Would you be willing to try out [safetydance-django](https://github.com/quansight/safetydance_django/)?

File: docker/docker-compose.yml (hunk 1 above)
Comment (xmnlab) [reply docker/docker-compose.yml]: wow you're right ... sorry ... I copied it from another file I have .. and I didn't see it was postgis ... I will replace that. thanks

File: scrud_django/tests/test_rest.py (hunk 2 above)
Comment (xmnlab) [reply scrud_django/tests/test_rest.py]: sounds good .. I will take a look how to use it :) thanks for the recommendation.

File: scrud_django/models.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+"""
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+
+class ResourceType(models.Model):
+    type_uri = models.URLField()
+    schema = models.ForeignKey(
+        'self',
```
Comment (trentoliphant) [inline scrud_django/models.py]: Change to reference "Resource" instead of self

File: scrud_django/models.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        null=True,
+        on_delete=models.PROTECT,
+        verbose_name=_('schema resource type'),
+        related_name='resource_type_schema',
+    )
+    context = models.ForeignKey(
+        'self',
```
Comment (trentoliphant) [inline scrud_django/models.py]: Same as above - reference "Resource" instead of self

File: scrud_django/models.py (hunk 3 above)
Comment (xmnlab) [reply scrud_django/models.py]: thanks @trentoliphant for catching that!

File: scrud_django/models.py (hunk 4 above)
Comment (xmnlab) [reply scrud_django/models.py]: thanks @trentoliphant for catching that!

**Comment by xmnlab** [comment]
@dcharbon this PR is ready again for review. note: CI is already working :)

**Review by dcharbon — APPROVED** [review]

**Comment by xmnlab** [comment]
some updates about this PR:

I updated the github actions for publishing the documentation.

It should work after merging. it will need a github token and application for that (https://stackoverflow.com/a/49007415/3715476). I don't have permissions to create that.

The github token secret name can be `GITHUB_TOKEN`.
The application name defined in the github actions was `scrud_django_app`.

thanks!

**Comment by xmnlab** [comment]
@trentoliphant do you have any other suggestion/recommendation about this PR?

**Pull request merged**